themes/bootstrap3/templates/Recommend/MapSelection.phtml

```diff
@@ -1,26 +1,38 @@
 <?php if ($this->recommend->getSearchResultCoordinates()) :?>
 <?php
-  $this->headScript()->appendFile('vendor/ol/ol.js');
-  $this->headScript()->appendFile('map_selection.js');
-  $this->headLink()->appendStylesheet('vendor/ol/ol.css');
+  $geoPlatform = $this->recommend->getGeoPlatform();
+  //Include files based on geoPlatform configuration
+  $this->headScript()->appendFile('vendor/leaflet/leaflet.js');
+  $this->headScript()->appendFile('vendor/leaflet/leaflet.draw.js');
+  $this->headScript()->appendFile('vendor/leaflet/leaflet.markercluster.js');
+  $this->headScript()->appendFile('map_selection_leaflet.js');
+  $this->headLink()->appendStylesheet('vendor/leaflet/leaflet.css');
+  $this->headLink()->appendStylesheet('vendor/leaflet/leaflet.draw.css');
+  $this->headLink()->appendStylesheet('vendor/leaflet/MarkerCluster.css');
+  $this->headLink()->appendStylesheet('vendor/leaflet/MarkerCluster.Default.css');
+  $this->headLink()->appendStylesheet('geofeatures.css');
 
-  $resultsCoords = $this->recommend->getMapResultCoordinates();
+  // Call methods based on geoPlatform configuration
   $basemap = $this->recommend->getBasemap();
-  $baseUrl = $this->url('home');
   $geoField = $this->recommend->getGeoField();
+  $height = $this->recommend->getHeight();
+  $showSelection = true;
+  $baseUrl = $this->url('home');
   $urlpath = $this->url('search-results');
+  $searchParams = $this->recommend->getSearchParams();
   $coordinates = $this->recommend->getSelectedCoordinates();
-  $showSelection = true;
-  $popupTitle = $this->transEsc('map_results_label');
+
   if ($coordinates == null) {
     $coordinates = $this->recommend->getDefaultCoordinates();
     $showSelection = false;
   }
-  $height = $this->recommend->getHeight();
-  $searchParams = $this->recommend->getSearchParams();
-  $params = [json_encode($geoField), json_encode($coordinates), json_encode($urlpath),
-    json_encode($baseUrl), json_encode($searchParams), json_encode($showSelection),
-    json_encode($resultsCoords), json_encode($popupTitle), json_encode($basemap)];
+  $helpTopic = 'geosearch_leaflet';
```

Comment:
Should we replace the existing help page rather than creating geosearch_leaflet, if we aren't keeping the old functionality anyway?

Reply:
Yes, I'll update that.